libnavigation-router/src/test/java/com/mapbox/navigation/route/internal/onboard/MapboxOnboardRouterTest.kt

```diff
@@ -206,6 +199,101 @@ class MapboxOnboardRouterTest {
         assertTrue(job.isCancelled)
     }
 
+    @Test
+    fun `cancel a request when multiple are running`() {
+        coEvery { navigator.getRoute(any()) } coAnswers {
+            awaitCancellation()
+        }
+
+        val firstLatch = CountDownLatch(1)
+        val firstCallback: Router.Callback = object : Router.Callback {
+            override fun onResponse(routes: List<DirectionsRoute>) {
+                Assert.fail()
```

Comment:
Same thing happens 👇 

```suggestion
                fail()
```

Reply:
Updated.